### PR TITLE
Support K8s 1.22/1.23 and update dependency versions

### PR DESCRIPTION
**Pull request by rjeberhard**
* Move all tests, documentation, and samples to use the "v1" API version of the Ingress resource
* Update Traefik and Nginx installs to latest versions that support "v1" Ingress resource
* Update dependencies to resolve OWASP vulnerabilities
* Update Kubernetes Java Client to 14.0.0 (PLS done)
* Simplify WeblogicApi to use CustomObjectsApi so that it will be much simpler to implement the Cluster resource support later
* Some rework to support more easily running the Main class in an IDE and debugger

Integration tests:
Kubernetes 1.19: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9114/
Kubernetes 1.21: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9110/
Kubernetes 1.22: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9111/
Kubernetes 1.23: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/9112/

### CODE REVIEW

**Review by sankarpn — APPROVED** [review]
Looks good to me.

**Review by ddsharpe — APPROVED** [review]

**Review by lennyphan — APPROVED** [review]
LGTM

**Review by ankedia — APPROVED** [review]
LGMT. Minor comments about couple of commented out lines that may need to be removed.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporterSideCar.java [hunk 1]
```diff
@@ -437,7 +433,7 @@ private void installPrometheusGrafana(String promChartVersion,
       if (OKD) {
         hostPortGrafana = createRouteForOKD(grafanaReleaseName, monitoringNS) + ":" + grafanaHelmParams.getNodePort();
       }
-      installVerifyGrafanaDashBoard(hostPortGrafana, monitoringExporterEndToEndDir);
+      // installVerifyGrafanaDashBoard(hostPortGrafana, monitoringExporterEndToEndDir);
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporterSideCar.java]: removed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporterWebApp.java [hunk 2]
```diff
@@ -436,7 +434,7 @@ private void installPrometheusGrafana(String promChartVersion,
       if (OKD) {
         hostPortGrafana = createRouteForOKD(grafanaReleaseName, monitoringNS) + ":" + grafanaHelmParams.getNodePort();
       }
-      installVerifyGrafanaDashBoard(hostPortGrafana, monitoringExporterEndToEndDir);
+      // installVerifyGrafanaDashBoard(hostPortGrafana, monitoringExporterEndToEndDir);
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporterWebApp.java]: removed

**Pull request merged**